trace-utils/src/send_data.rs

```diff
@@ -5,83 +5,171 @@ use anyhow::{anyhow, Context};
 use bytes::Bytes;
 use futures::stream::FuturesUnordered;
 use futures::StreamExt;
-use hyper::{Body, Client, Method, Response, StatusCode};
+use hyper::{Body, Client, Method, Response};
 use std::collections::HashMap;
+// use std::fmt::Result;
 use std::time::Duration;
 use tokio::time::sleep;
 
 use crate::tracer_header_tags::TracerHeaderTags;
-use datadog_trace_protobuf::pb;
+use datadog_trace_protobuf::pb::{self, TracerPayload};
 use ddcommon::{connector, Endpoint, HttpRequestBuilder};
 
+const DD_API_KEY: &str = "DD-API-KEY";
+const HEADER_DD_TRACE_COUNT: &str = "X-Datadog-Trace-Count";
+const HEADER_HTTP_CTYPE: &str = "Content-Type";
+const HEADER_CTYPE_MSGPACK: &str = "application/msgpack";
+const HEADER_CTYPE_PROTOBUF: &str = "application/x-protobuf";
+
+type BytesSent = u64;
+type ChunksSent = u64;
+type ChunksDropped = u64;
+type Attempts = u32;
+
 #[derive(Debug)]
-pub enum SendRequestError {
-    Hyper(hyper::Error),
-    Any(anyhow::Error),
+enum RequestError {
+    Build,
+    Network,
+    Timeout,
+}
+
+impl std::fmt::Display for RequestError {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        match self {
+            RequestError::Timeout => write!(f, "Connection timed out"),
+            RequestError::Network => write!(f, "Network error"),
+            RequestError::Build => write!(f, "Request failed due to invalid property"),
+        }
+    }
 }
 
+impl std::error::Error for RequestError {}
+
+pub enum RequestResult {
+    // Holds information from a succesful request.
```

Comment:
These should probably be `///` instead of `//` so they're treated as doc comments?